File: Challenge/FIZZBUZZ/fizzbuzz.dart
void main() {
  fizzBuzz(100);
}

void fizzBuzz(int number) {
  ///
  for (int i = 1; i <= number; i++) {
    if (i % 15 == 0) {
      // 3*5 = 15
      print('FizzBuzz');
    } else if (i % 5 == 0) {
      // 5
      print('Buzz');
    } else if (i % 3 == 0) {
      // 3
      print('Fizz');
    } else {
      print(i);
    }
  }

  print('Done');
}

File: Challenge/FIZZBUZZ/fizzbuzz.ts
const FizzBuzz = (num: number): string => {
  let result: number;

  for (let i = 1; i <= num; i++) {
    if (i % 3 === 0 && i % 5 === 0) {
      result = i;
      console.log("FizzBuzz");
    }
    if (i % 3 === 0) {
      result = i;
      console.log("Fizz");
    }

    if (i % 5 === 0) {
      result = i;
      console.log("Buzz");
    }

    if (i % 3 !== 0 && i % 5 !== 0) {
      result = i;
      console.log(result);
    }
  }

  return "Done";
};

FizzBuzz(100);
